### ISSUE TITLE
Optionally support focusable disabled elements

### ISSUE THREAD

**Issue by MarvinPerry** [Esri/calcite-design-system#12547]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

At the moment, calcite-components use the native disabled attribute, which makes elements non-interactive as expected, but that also removes them from the focus order, which makes them inaccessible to users that rely on screen readers or that navigate by keyboard.

We’re proposing a change to how we handle “disabled” states for interactive elements (buttons, menu items and such) both visually and functionally. Instead we’d like these elements to be styled with a .disabled class (with pointer-events: none) and use aria-disabled="true" for accessibility. This would allow elements to remain focusable, so that users can still access tooltips or other contextual guidance explaining why the element is unavailable.

<img width="500" height="1548" alt="Image" src="https://github.com/user-attachments/assets/6b209515-9c63-4e13-a69f-665c5df22ddc" />
<img width="500" height="1912" alt="Image" src="https://github.com/user-attachments/assets/37ca64e6-7d5a-40fc-9fae-cf4583f04ba8" />

### Acceptance Criteria

A `.disabled` class is added that visually matches our current disabled styling.

- Elements using `.disabled` should:
    - have `pointer-events: none` or a similar technique to prevent interaction.
    -  have `aria-disabled="true"` to indicate the disabled state to assistive tech.
    -  remain keyboard focusable and not use the `disabled` attribute, so they remain in the dom

### Relevant Info

This change supports accessibility improvements, specifically for screen reader and keyboard-only users that can't currently access tooltips associated with the disabled elements.

### Which Component

N/A

### Example Use Case

An extremely simple example of our ask:
https://codepen.io/MarvinPerry/pen/qEObOob

### Current disabled button
```html
<calcite-button id="my-button" disabled title="with disable attribute">Click me</calcite-button>

<calcite-tooltip reference-element="my-button"> <span>screen readers won't read this</span> </calcite-tooltip>
```

### Proposed disabled button
```html
<calcite-button id="my-button" class="disabled" aria-disabled="true" title="with disabled class">Click me</calcite-button>

<calcite-tooltip reference-element="my-button"> <span>screen readers will read this</span> </calcite-tooltip>


### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [x] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**Comment by geospatialem** [comment]
@MarvinPerry Calcite follows native `disabled` functionality, however the above could be achieved preventing the default behavior and some context provided both visually and to assistive technologies. 

Would setting different attributes/properties work for the use case, where the `button` isn't `disabled`, but context is provided visually and through assistive technologies using the component's `kind`, `label`, preventing the default behavior, and toggling the `tooltip`'s `hidden` attributes?: https://codepen.io/geospatialem/pen/xbwZrNV

![Image](https://github.com/user-attachments/assets/52b73865-90ce-4d27-92b1-6b4989acc120)

**Comment by juliannemarik** [comment]
Hi @geospatialem - thank you for the reply and suggestion above, and apologies for our slow response.

> Calcite follows native disabled functionality

Hmm, we may have implemented something incorrectly here, but we are not observing this 🤔 

[Here's](https://codepen.io/juliannemarik/pen/EaVdxeY) a codepen for reference - when using VO on Safari, I am able to navigate to/focus on the disabled native button, and the associated tooltip text gets read. Conversely, the disabled Calcite button is not focusable, and the tooltip text does not get read. Is this difference expected or are we doing something incorrectly? 

https://github.com/user-attachments/assets/d032e97c-feac-4abd-82b8-e6d6a5a003f6

**Comment by geospatialem** [comment]
> [@MarvinPerry](https://github.com/MarvinPerry) Calcite follows native `disabled` functionality, however the above could be achieved preventing the default behavior and some context provided both visually and to assistive technologies.
> 
> Would setting different attributes/properties work for the use case, where the `button` isn't `disabled`, but context is provided visually and through assistive technologies using the component's `kind`, `label`, preventing the default behavior, and toggling the `tooltip`'s `hidden` attributes?: https://codepen.io/geospatialem/pen/xbwZrNV
> 
> ![Image](https://github.com/user-attachments/assets/52b73865-90ce-4d27-92b1-6b4989acc120) [ ![Image](https://github.com/user-attachments/assets/52b73865-90ce-4d27-92b1-6b4989acc120) ](https://private-user-images.githubusercontent.com/5023024/468306926-52b73865-90ce-4d27-92b1-6b4989acc120.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5MTU3ODcsIm5iZiI6MTc1NjkxNTQ4NywicGF0aCI6Ii81MDIzMDI0LzQ2ODMwNjkyNi01MmI3Mzg2NS05MGNlLTRkMjctOTJiMS02YjQ5ODlhY2MxMjAuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MDNUMTYwNDQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGNiMWI2M2Q5ZDJkZmQyN2QwN2M1NjlkZGE4YThmMWMwNDE2OTc0ZDVjNjFkZDZlZmI3NjQyZjllZjQ0YTVlNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4NlH-joufLnHAl3jafTGpqrcLWtN1xt5nFb3trZ0IHY) [ ](https://private-user-images.githubusercontent.com/5023024/468306926-52b73865-90ce-4d27-92b1-6b4989acc120.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5MTU3ODcsIm5iZiI6MTc1NjkxNTQ4NywicGF0aCI6Ii81MDIzMDI0LzQ2ODMwNjkyNi01MmI3Mzg2NS05MGNlLTRkMjctOTJiMS02YjQ5ODlhY2MxMjAuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MDNUMTYwNDQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGNiMWI2M2Q5ZDJkZmQyN2QwN2M1NjlkZGE4YThmMWMwNDE2OTc0ZDVjNjFkZDZlZmI3NjQyZjllZjQ0YTVlNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4NlH-joufLnHAl3jafTGpqrcLWtN1xt5nFb3trZ0IHY)

Closing the above as unplanned, @MarvinPerry please reach out if the above use case doesn't fit the need for re-evaluation.
